Lonely/Lonely/GameLib/DirectX/DirectX.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+* @file	Direct3D.h
+* @brief Direct3D�N���X�̃w�b�_�t�@�C��
+* @author ����@�k��
+*/
+
+#include "DirectX.h"
+
+DirectX::DirectX()
+	: m_pD3D(NULL)
+	, m_pDevice(NULL)
+{
+}
+
+DirectX::~DirectX()
+{
+	Finalize();
+}
+
+
+// DirectX�̏���
+bool DirectX::Initialize(HWND hWnd, int width, int height, bool isFullscreen)
+{
+	if (m_pD3D != NULL || m_pDevice != NULL) return false;
+
+	// IDirect3D9�I�u�W�F�N�g�̍쐬
+	if ((m_pD3D = Direct3DCreate9(D3D_SDK_VERSION)) == NULL)
+	{
+		return false;
+	}
+
+	D3DPRESENT_PARAMETERS d3dpp;
+	ZeroMemory(&d3dpp, sizeof(D3DPRESENT_PARAMETERS));
+	// Device�̐ݒ�
+	d3dpp.Windowed = !isFullscreen;
+	d3dpp.hDeviceWindow = hWnd;
+	d3dpp.BackBufferWidth = width;
+	d3dpp.BackBufferHeight = height;
+	d3dpp.BackBufferFormat = D3DFMT_UNKNOWN;
+	d3dpp.SwapEffect = D3DSWAPEFFECT_DISCARD;
+	d3dpp.MultiSampleType = D3DMULTISAMPLE_8_SAMPLES;//D3DMULTISAMPLE_NONE;//D3DMULTISAMPLE_8_SAMPLES;
+	d3dpp.MultiSampleQuality = 0;
+	d3dpp.Flags = 0;
+	d3dpp.BackBufferCount = 1;
+	d3dpp.EnableAutoDepthStencil = true;
+	d3dpp.AutoDepthStencilFormat = D3DFMT_D24S8;
+	// VSYNC�҂��L��)
+	d3dpp.PresentationInterval = D3DPRESENT_INTERVAL_ONE;
+	d3dpp.FullScreen_RefreshRateInHz = D3DPRESENT_RATE_DEFAULT;
+
+	// Direct3D�f�o�C�X�̐���
+	if (FAILED(m_pD3D->CreateDevice(D3DADAPTER_DEFAULT,
+		D3DDEVTYPE_HAL,
+		d3dpp.hDeviceWindow,
+		D3DCREATE_HARDWARE_VERTEXPROCESSING | D3DCREATE_MULTITHREADED,
+		&d3dpp,
+		&m_pDevice)))
+	{
+		SAFE_RELEASE(m_pD3D);
+		return false;
+	}
+	m_d3dpp = d3dpp;
+	return true;
+}
+
+// DirectX�̏I������
+void DirectX::Finalize()
+{
+	SAFE_RELEASE(m_pDevice);
+	SAFE_RELEASE(m_pD3D);
+}
+
+
+// �`��V�[���̊J�n
+void DirectX::BeginRenderScene()
+{
+	if (m_pDevice) m_pDevice->BeginScene();
```

Comment:
C/C++的にはなんら問題無いのですが、
```suggestion
	if (m_pDevice != NULL) m_pDevice->BeginScene();
```
で慣れておいた方が良いような気がします。

Reply:
場所によって書き方がマチマチなので再指摘しておきました。